MON-3940: Add the collection of MTV migration metrics to Telemetry

bkhizgiy opened the pull request:
Following this JIRA issue https://issues.redhat.com/browse/MON-3940

Related to this PRs for adding metrics on the MTV side:
https://github.com/kubev2v/forklift/pull/916
https://github.com/kubev2v/forklift/pull/932
https://github.com/kubev2v/forklift/pull/978

fabiand commented:
@simonpasquier please review

simonpasquier reviewed (commented):
You'd need to configure recording rules aggregating on the allowed labels: 
* status
* provider
* mode
* target

If the metrics are emitted by a single container then you can do:

```
record: cluster:mtv_migrations_status_total:max
expr: max by(status, provider, mode, target) (mtv_migrations_status_total)
````

otherwise you should probably sum

```
record: cluster:mtv_migrations_status_total:sum
expr: sum by(status, provider, mode, target) (mtv_migrations_status_total)
````

simonpasquier commented:
/retitle MON-3940: Add the collection of MTV migration metrics to Telemetry
/hold

until explicit approval

bkhizgiy commented:
@simonpasquier Thanks for the review. Should the recording rule be part of the MTV code, or can we specify the expression in the whitelist? This metric consists only of the mentioned fields, so there aren't any labels to remove or modify.

fabiand commented:
Is anything missing to get this merged?

fabiand commented:
@simonpasquier any update?

simonpasquier commented:
> Should the recording rule be part of the MTV code, or can we specify the expression in the whitelist? 

MTV code. The telemetry allow-list only references metric names (and labels).

> This metric consists only of the mentioned fields, so there aren't any labels to remove or modify.

We still ask to remove the labels that are useless to keep at the Telemetry server level like `instance` and `pod`.

simonpasquier commented:
see https://rhobs-handbook.netlify.app/products/openshiftmonitoring/telemetry.md/#configure-recording-rules

juzhao commented on Documentation/data-collection.md:
```diff
@@ -257,6 +257,12 @@ data:
+    #
+    # cluster:mtv_migrations_status_total is the total number of VM migrations running on the cluster,
+    # labeled with {status}, {provider}, {mode}, and {target}.
+    - '{__name__="mtv_migrations_status_total"}'
```
the metrics name `mtv_migrations_status_total` is not the same as `cluster:mtv_migrations_status_total` in
`cluster:mtv_migrations_status_total is the total number of VM migrations running on the cluster,`
maybe should be `cluster:mtv_migrations_status_total:sum` as @simonpasquier mentioned in https://github.com/openshift/cluster-monitoring-operator/pull/2461#pullrequestreview-2292267082

bkhizgiy commented:
Following the comments, I’ve added a new recording rule for MTV in this [PR](https://github.com/kubev2v/forklift/pull/1172).
I’ve also updated this PR with the new rule name accordingly. 
Please let me know if anything else is needed.

fabiand commented:
https://github.com/kubev2v/forklift/pull/1172 got merged - what is needed to move this PR forward?

sradco commented:
Hi @bkhizgiy, looks like the PR needs a rebase. Is there anything else keeping thos PR from getting merged?

juzhao commented:
the PR is outdated and replaced by https://github.com/openshift/cluster-monitoring-operator/pull/2743
/close
